src/Compilers/CSharp/Portable/Symbols/PublicModel/PropertySymbol.cs

```diff
@@ -107,6 +107,8 @@ ImmutableArray<CustomModifier> IPropertySymbol.RefCustomModifiers
 
         bool IPropertySymbol.ReturnsByRefReadonly => _underlying.ReturnsByRefReadonly;
 
+        ISymbol IPropertySymbol.ContainingSymbol => _underlying.ContainingSymbol.GetPublicSymbol();
```

Comment:
i'm curious if these can be more strongly typed.  e.g. should the ContainingSymbol for an IPropertySymbol just be an INamedTyepSymbol?

Reply:
A change of that level feels like it would need to be deeper, go through API review, etc ... This was more addressed at putting the pain of NRT in compiler vs. the IDE team. This approach removed the necessity of adding lots of `!` and `Debug.Assert` in the IDE layer. 

Note: I sent this PR to have a discussion with compiler team about "should we take this approach or not". So it's not in it's final form, wanted it mostly as a tool for discussion.

Reply:
understaood.  but once we add these here, we can't do the covariant stuff later.  so i'd def want it to be part of the dsicussion we have here before we make a change like this :)  thanks!

Reply:
I experimented with this idea locally for a bit and

1. It's a much more invasive change. The change would be a lot easier I think if we could use covariant returns but sadly we can't. The remaining tools at our disposal make the code less appealing. It also gets pretty deep into the compiler because we end up needing to plumb this all the way through at least the retargeting layer. That's where I stopped digging so not sure if it goes deeper. 
2. It's possible this would impact customers in a negative way. Consider that `var symbol = prop.ContainingSymbol` would now infer a different type. One that may cause errors further down the line if say they later re-assigned this with an expression that was simply `ISymbol`. 

I think this suggestion could be valuable but it's a lot of work and has non-trivial customer impact. Definitely needs to go through API review and possibly compat discussions. As such think I'm going to continue on the current path and if we decide to invest in this more strongly typed rout we can adjust the definitions I'm adding.

Reply:
> I think this suggestion could be valuable but it's a lot of work and has non-trivial customer impact. Definitely needs to go through API review and possibly compat discussions. 

definitely.  htat's what i'm suggesting :)

> and if we decide to invest in this more strongly typed rout we can adjust the definitions I'm adding.

agreed. i  just didn't want us to make this change without doing that work as it would likely mean not being able to do it in the future :)